core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -16,203 +16,252 @@
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+  private _changeStoryPublicationStatus(
+      storyId: string,
+      newStoryStatusIsPublic: boolean,
+      successCallback: (value: number) =>void,
+      errorCallback: (reason: string) => void): void {
+    const storyPublishUrl = this.urlInterpolationService.interpolateUrl(
+      StoryDomainConstants.STORY_PUBLISH_URL_TEMPLATE, {
+        story_id: storyId
       });
+    const putData = {
+      new_story_status_is_public: newStoryStatusIsPublic
     };
+    // eslint-disable-next-line max-len
+    this.http.put(storyPublishUrl, putData, { observe: 'response'}).toPromise().then(
+      response => successCallback(response.status),
+      errorResponse => errorCallback(errorResponse.error.error));
+  }
 
-    var _validateExplorations = function(
-        storyId, expIds, successCallback, errorCallback) {
-      var validateExplorationsUrl = UrlInterpolationService.interpolateUrl(
-        VALIDATE_EXPLORATIONS_URL_TEMPLATE, {
+  private _validateExplorations(
+      storyId: string,
+      expIds: string[],
+      successCallback: (value: ValidationErrorMessages) => void,
+      errorCallback: (reason: string) => void): void {
+    const validateExplorationsUrl =
+      this.urlInterpolationService.interpolateUrl(
+        StoryDomainConstants.VALIDATE_EXPLORATIONS_URL_TEMPLATE, {
           story_id: storyId
         });
 
-      $http.get(validateExplorationsUrl, {
+    this.http.get<ValidationErrorMessages>(
+      validateExplorationsUrl, {
         params: {
           comma_separated_exp_ids: expIds.join(',')
         }
-      }).then(function(response) {
-        if (successCallback) {
-          successCallback(response.data.validation_error_messages);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+      }
+    ).toPromise().then(
+      response => successCallback(response.validation_error_messages),
+      errorResponse => errorCallback(errorResponse.error.error));
+  }
+
+  private _deleteStory(
+      storyId: string,
+      successCallback: (value: void) =>void,
+      errorCallback: (reason: string) => void): void {
+    const storyDataUrl = this.urlInterpolationService.interpolateUrl(
+      StoryDomainConstants.EDITABLE_STORY_DATA_URL_TEMPLATE, {
+        story_id: storyId
       });
-    };
 
-    var _deleteStory = function(
-        storyId, successCallback, errorCallback) {
-      var storyDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_STORY_DATA_URL_TEMPLATE, {
-          story_id: storyId
-        });
-      $http['delete'](storyDataUrl).then(function(response) {
+    // eslint-disable-next-line
+    this.http.delete<void>(
+      storyDataUrl, {observe: 'response'}).toPromise().then(
+      (response) => {
         if (successCallback) {
-          successCallback(response.status);
+          successCallback();
```

Comment:
Why did you remove `response.status` here?

Reply:
corrected